engine/chain/internal/bhs/bhs_error_response.go

```diff
@@ -0,0 +1,57 @@
+package bhs
+
+import (
+	"encoding/json"
+	"net/http"
+	"strconv"
+
+	"github.com/bitcoin-sv/spv-wallet/engine/chain/errors"
+	"github.com/bitcoin-sv/spv-wallet/engine/spverrors"
+	"github.com/bitcoin-sv/spv-wallet/models"
+	"github.com/go-resty/resty/v2"
+)
+
+// mapBHSErrorResponseToSpverror is a method that will check what kind of response came back from
+// Block Header Service and map it to spverror and set it to context
+func mapBHSErrorResponseToSpverror(res *resty.Response) models.SPVError {
+	var responseErr bhsError
+
+	if err := json.Unmarshal(res.Body(), &responseErr); err != nil {
+
+		// TODO: remove after SPV-1106 is done
```

Comment:
Please add appropriate task to jira to not forget about this `TODO`.
- or point out in `SPV-1106` description to do so 😄

Reply:
after discussion, removed the comment and left this handle error here, It might be useful when server returns 404 not foundd which is a string by default